core/src/main/java/com/ritense/valtimo/service/CamundaTaskService.java

```diff
@@ -161,13 +163,22 @@ public CamundaTask findTaskById(String taskId) {
             .orElseThrow(() -> new TaskNotFoundException(String.format("Cannot find task %s", taskId)));
     }
 
+    @Transactional
+    public void assignByEmail(String taskId, String assigneeEmail) throws IllegalStateException {
+        var assignee = userManagementService.findNamedUserByEmail(assigneeEmail)
+            .orElseThrow(() -> new IllegalStateException("Error. No registered user found with email: " + assigneeEmail));
+        assign(taskId, assignee.getId());
+    }
+
     @Transactional
     public void assign(String taskId, String assignee) throws IllegalStateException {
         if (assignee == null) {
             unassign(taskId);
+        } else if (assignee.matches("[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\\.[a-zA-Z]{2,4}")) {
```

Comment:
I can't use the keycloak validation here. But this extra check is only there for developers that forgot to change the assign function to now use a keycloak UUID instead of an email.

Reply:
You could instead use the `EmailValidator` from Apache commons, or the regex referenced by [OWASP](https://owasp.org/www-community/OWASP_Validation_Regex_Repository):
`^[a-zA-Z0-9_+&*-]+(?:\\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\\.)+[a-zA-Z]{2,}$`